src/palindrom.js

```diff
@@ -303,21 +303,30 @@ const Palindrom = (() => {
          * @param  {String} msg message to be sent
          * @return {PalindromNetworkChannel}     self
          */
-        send(msg) {
+        async send(msg) {
             // send message only if there is a working ws connection
             if (this.useWebSocket && this._ws && this._ws.readyState === 1) {
                 this._ws.send(msg);
                 this.onSend(msg, this._ws.url, 'WS');
             } else {
                 const url = this.remoteUrl.href;
-                this.xhr(
-                    url,
-                    'application/json-patch+json',
-                    msg,
-                    (res, method) => {
-                        this.onReceive(res.data, url, method);
-                    }
-                );
+                try {
+                    const res = await this.xhr(
+                        url,
+                        'application/json-patch+json',
+                        msg
+                    );
+
+                    this.handleLocationHeader(res);
+
+                    this.onReceive(
+                        res.data,
+                        url,
+                        res.config.method.toUpperCase()
+                    );
+                } catch (error) {
+                    this.handleFailureResponse(error);
+                }
```

Comment:
What is the reason for moving `handleLocationHeader` and `handleFailureResponse` from `xhr`?
I think it's safer to keep it there, as `handleLocationHeader` is called every time after you call and resolve `xhr`. The fact `handleFailureResponse` is sometimes not called looks to me as a bug.

Reply:
If ``handleFailureResponse`` would be a part of `xhr` you would not need `try catch` here and in many other places